controllers/pod_node_module_reconciler.go

```diff
@@ -0,0 +1,166 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/qbarrand/oot-operator/internal/constants"
+	"github.com/qbarrand/oot-operator/internal/daemonset"
+	"github.com/qbarrand/oot-operator/internal/filter"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubectl/pkg/util/podutils"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+//+kubebuilder:rbac:groups="core",resources=pods,verbs=get;patch;list;watch
+//+kubebuilder:rbac:groups="core",resources=nodes,verbs=get;watch
+
+type PodNodeModuleReconciler struct {
+	client client.Client
+}
+
+func NewPodNodeModuleReconciler(client client.Client) *PodNodeModuleReconciler {
+	return &PodNodeModuleReconciler{client: client}
+}
+
+func (pnmr *PodNodeModuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	pod := v1.Pod{}
+	podNamespacedName := req.NamespacedName
+
+	if err := pnmr.client.Get(ctx, podNamespacedName, &pod); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("Pod not found")
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, fmt.Errorf("could not get pod %s: %v", podNamespacedName, err)
+	}
+
+	nodeName := pod.Spec.NodeName
+
+	moduleName, ok := pod.Labels[constants.ModuleNameLabel]
+	if !ok {
+		return ctrl.Result{}, fmt.Errorf("pod %s has no %q label", podNamespacedName, constants.ModuleNameLabel)
+	}
+
+	labelName := daemonset.GetDriverContainerNodeLabel(moduleName)
+
+	deleting := !pod.DeletionTimestamp.IsZero()
+
+	logger = logger.WithValues(
+		"node name", nodeName,
+		"module name", moduleName,
+		"label name", labelName,
+		"deleting pod", deleting,
+	)
+
+	if !podutils.IsPodReady(&pod) || deleting {
```

Comment:
i think you can remove deleting condition here: you are checking it anyway in line73, and pod cannot be both ready and deleting

Reply:
Does deleting automatically remove the `Ready` condition? I could not find any documentation on this.

Reply:
Delete is set by pod controller when the containers are deleted. Once the containers are deleted, pod cannot be in Ready state. But even if we have a race condition here, we will not fail, since we will catch it on the next reconciliation loop

Reply:
Indeed. Simplified the logic.